tests/lib/context/basic_test.js

```diff
@@ -113,15 +158,49 @@ describe('A simple context with one resource', function() {
 
     it('should return the correct plural form for 1', function(done) {
       var args = {unread: 1};
-      ctx.resolveValue(langs, 'unreadMessages', args).then(function(value) {
-        assert.strictEqual(value, 'One unread');
-      }).then(done, done);
+      ctx.resolveValues(langs, [['unreadMessages', args]]).then(
+        function([value]) {
+          assert.strictEqual(value, 'One unread');
+        }
+      ).then(done, done);
     });
 
     it('should return for missing string', function(done) {
-      ctx.resolveValue(langs, 'missingId').then(function(value) {
+      ctx.resolveValues(langs, ['missingId']).then(function([value]) {
         assert.strictEqual(value, 'missingId');
       }).then(done, done);
     });
+
+    it('should return for many', function(done) {
+      ctx.resolveValues(langs, ['brandName', 'about']).then(
+        function([brandName, about]) {
+          assert.strictEqual(brandName, 'Firefox');
+          assert.strictEqual(about, i('About Firefox', 'Firefox'));
+        }
+      ).then(done, done);
+    });
+
+    it('should return for many including a missing one', function(done) {
+      ctx.resolveValues(langs, ['brandName', 'missingId', 'about']).then(
+        function([brandName, missingId, about]) {
+          assert.strictEqual(brandName, 'Firefox');
+          assert.strictEqual(missingId, 'missingId');
+          assert.strictEqual(about, i('About Firefox', 'Firefox'));
+        }
+      ).then(done, done);
+    });
+
+    it('should return for many including an id/args pair', function(done) {
+      var args = {unread: 1};
+      ctx.resolveValues(langs, [
+        'brandName',
+        ['unreadMessages', args],
```

Comment:
Could you add one more key here with the same id?

    ['unreadMessages', {unread: 1}],
    ['unreadMessages', {unread: 2}],